src/worker/vm/imports.ts

```diff
@@ -2,14 +2,76 @@ import { BigQuery } from '@google-cloud/bigquery'
 import * as contrib from '@posthog/plugin-contrib'
 import * as AWS from 'aws-sdk'
 import crypto from 'crypto'
+import dns from 'dns'
 import * as genericPool from 'generic-pool'
-import fetch from 'node-fetch'
-import * as pg from 'pg'
+import nodeFetch from 'node-fetch'
+import nodePostgres from 'pg'
+import { parse } from 'pg-connection-string'
 import snowflake from 'snowflake-sdk'
+import url from 'url'
 import * as zlib from 'zlib'
 
 import { writeToFile } from './extensions/test-utils'
 
+// 93.184.216.34 = example.com
+const RESTRICTED_IPS = ['93.184.216.34']
+
+const isIpv4 = (str: string) =>
+    /^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/.test(
+        str
+    )
+const isIpv6 = (str: string) =>
+    /(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))/.test(
+        str
+    )
+
+const validateHostOrUrl = async (hostOrUrl: any) => {
+    if (typeof hostOrUrl != 'string') {
+        throw new Error(`Invalid host/URL ${hostOrUrl}`)
+    }
+    if (
+        (hostOrUrl.includes('amazonaws.com') && !hostOrUrl.includes('redshift.amazonaws.com')) ||
+        hostOrUrl.includes('localhost') ||
+        hostOrUrl.includes('posthog.net') ||
```

Comment:
I suggest doing these `hostOrUrl` lookups on the `parsedHost` parameter below. This way we can use exact match and avoid issues with `posthog.netsomeclient.com`

Reply:
Hmm this lingered from the previous changes but I guess amazon checks could go below. IP checks and localhost should happen here though because of `url.parse`:

```
url.parse('localhost:8000').hostname // '8000'
url.parse('127.0.0.1').hostname // null
```
Having written this though I guess IPs could be checked via a null hostname?

Reply:
Ok so IP addresses following IPv4 return null hostnames, IPv6 addresses don't.

Kept `localhost` and IPv6 here and moved the rest under